### DISCUSSION TITLE
Telling whisper to ignore certain words (similar to initial_prompt)

### DISCUSSION THREAD

**Discussion by Abdurrafey-Siddiqui** [openai/whisper#374] [Q&A] [answered]
We know that initial_prompt makes whisper incline to a word while transcribing, now what if i want the opposite where I want whisper to ignore that word and not transcribe it. So basically similar to initial_prompt but the opposite outcome.

**Comment by jianfch** [comment, marked as answer]
```python
import whisper
tokenizer = whisper.tokenizer.get_tokenizer(multilingual=True) 
# multilingual=False if using .en models
token_to_be_suppressed = tokenizer .encode(' '+'these words will be suppressed'.strip()) + [-1]  
# -1 is there to suppress the special characters, which is the default behavior 

model = load_model('base')
results = model.transcribe(..., suppress_tokens=token_to_be_suppressed )
```

**Reply by Abdurrafey-Siddiqui** [reply]
@jianfch thanks for the answer

I tried this but it doesn't seem to be ignoring the words

**Reply by jianfch** [reply]
Likely because a word can have multiple combination of tokens and `tokenizer.encode` merely gives one of those combination. A simpler way would be to just throw out the words during postprocessing.

**Reply by Abdurrafey-Siddiqui** [reply]
ok no worries
